packages/block-library/src/footnotes/format.js

```diff
@@ -86,10 +97,23 @@ export const format = {
 					onChange( newValue );
 				}
 
+				// Attempt to find a common parent post content block.
+				// This allows for locating blocks within a page edited in the site editor.
+				const parentPostContent = getBlockParentsByBlockName(
```

Comment:
Nit: we could be using the found root client id below when inserting a footnotes if none is present.